doc/testing.md

```diff
@@ -0,0 +1,186 @@
+| `udp-push-pop`   | Tests for fixed size, unidirectional data transfers using UDP.            |
+| `udp-ping-pong`  | Tests for fixed size, bi-directional, symmetric data transfers using UDP. |
+
+**The following system-level tests are supported by `catmem`.**
+
+| Test Name        | Description                                                      |
+|------------------|------------------------------------------------------------------|
+| `pipe-push-pop`  | Tests for fixed size, unidirectional data transfers.             |
+| `pipe-ping-pong` | Tests for fixed size, bi-directional, symmetric data transfers.  |
+
+## How to Run System-Level Tests
+
+System-level tests may be run in multiple ways:
+
+- [Using the Demikernel's CI tool (ie. `demikernel-ci`)](#running-system-level-tests-with-demikernels-ci-tool-ie-demikernel-ci). This is the preferred method.
+- [Using Demikernel's build system (ie. `make`)](#running-system-level-tests-with-demikernels-build-system-ie-make). This method is discouraged, because you will have to deal by yourself with specific arguments for each test.
+- [Directly from the shell (ie. `bash`)](#running-system-level-tests-from-the-shell-ie-bash). This method is strongly discouraged, because you will have to not only deal with
+specific arguments for each test, but also with intricacies of your shell.
+
+> Regardless the method you choose to follow, you should have at your disposal a pair of machines with Demikernel's
+development environment set up. For more information on how to set up Demikernel's development environment, check out
+instructions in the `README.md` file.
+
+### Running System-Level Tests with Demikernel's CI Tool (ie. `demikernel-ci`)
+
+In order to use this method, you should first do the following setup once:
+
+- Setup RSA-based authentication for both server and host machines.
+- Setup SSH aliases for connecting to both server and host machines.
+
+```bash
+# Set parameters for remote machines.
+export SERVER_HOSTNAME=demikernel-vm0
+export SERVER_IPV4_ADDR=10.9.1.10
+export CLIENT_HOSTNAME=demikernel-vm1
+export CLIENT_IPV4_ADDR=10.9.1.11
+export DEMIKERNEL_PATH=/path/to/demikernel
+export DEMIKERNEL_BRANCH=dev
+
+# Run system-level tests with Catnap LibOS, unless otherwise stated.
+# If you want to run unit tests on a different LibOS, then set the LIBOS flag accordingly.
+export LIBOS=catnap
+
+# Run all system tests for the target LIBOS.
+python3 tools/demikernel-ci.py \
+    --server $SERVER_HOSTNAME \
+    --client $CLIENT_HOSTNAME \
+    --repository $DEMIKERNEL_PATH \
+    --branch $DEMIKERNEL_BRANCH \
+    --libos $LIBOS \
+    --test \
+    --server-addr $SERVER_IPV4_ADDR \
+    --client-addr $CLIENT_IPV4_ADDR
+```
+
+### Running System-Level Tests with Demikernel's Build System (ie. `make`)
+
+```bash
+# Set parameters for Demikernel's TCP/UDP stack.
+export SERVER_IPV4_ADDR=10.9.1.10:12345
+export CLIENT_IPV4_ADDR=10.9.1.11:12345
+
```

Comment:
For completeness, I'll note that the IPv4 private address ranges specified in RFC 1918 are [specifically allowed to be used in documentation](https://www.rfc-editor.org/rfc/rfc5737), so we're okay using these 10/8 prefix addresses here.

There is also space specifically reserved for documentation, including three /24 prefixes for IPv4, but I don't think we need to bother with those.

Reply:
Thanks for noting this. I changed the instructions to use the `192.0.2.0/24` address range.